### ISSUE TITLE
Problem - Optimizer hyper-parameter assignment 

### ISSUE THREAD

**Issue by miranthajayatilake** [keras-team/keras#9551]
Hi,
I'm using Keras to model GAN algorithms and I was testing with different variations of optimizers.
When first assigning the values I used the format:
`optimizer = Adam(0.0002, 0.5)`
Then I changed the format to:
`optimizer = Adam(lr=0.0002, beta_1=0.5)`

Surprisingly I found the two outputs to be somewhat different. As far as I can understand, since the function parameters are in the order as `keras.optimizers.Adam(lr=0.001, beta_1=0.9, beta_2=0.999, epsilon=None, decay=0.0, amsgrad=False)` which is shown in the documentation, the above two formats should be exactly the same assigning 0.0002 to the learning rate and 0.2 to the beta_1 parameters.

Has anyone else noticed this as well? If so what can be the reason for this behavior?

Thanks in advance!
